src/Compilers/VisualBasic/Test/Emit/PrivateProtected.vb

```diff
@@ -0,0 +1,567 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Collections.Immutable
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.VisualBasic.UnitTests.Emit
+Imports Roslyn.Test.Utilities
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests
+    Public Class PrivateProtected
+        Inherits BasicTestBase
+
+        Private Shared ReadOnly s_defaultProvider As DesktopStrongNameProvider = New SigningTestHelpers.VirtualizedStrongNameProvider(ImmutableArray.Create(Of String)())
+
+        <Fact>
+        Public Sub RejectIncompatibleModifiers()
+            Dim compilation = CreateCompilationWithMscorlibAndVBRuntime(
+                    <compilation>
+                        <file name="a.vb">
+                            <![CDATA[
+Public Class Base
+    Private Friend Field1 As Integer
+    Friend Private Field2 As Integer
+    Private Friend Protected Field3 As Integer
+    Friend Protected Private Field4 As Integer
+End Class
+]]>
+                        </file>
+                    </compilation>,
+                    parseOptions:=TestOptions.Regular.WithLanguageVersion(LanguageVersion.VisualBasic15_5))
+            CompilationUtils.AssertTheseDiagnostics(compilation,
+<errors>
+BC30176: Only one of 'Public', 'Private', 'Protected', 'Friend', or 'Protected Friend' can be specified.
+    Private Friend Field1 As Integer
+            ~~~~~~
+BC30176: Only one of 'Public', 'Private', 'Protected', 'Friend', or 'Protected Friend' can be specified.
+    Friend Private Field2 As Integer
+           ~~~~~~~
+BC30176: Only one of 'Public', 'Private', 'Protected', 'Friend', or 'Protected Friend' can be specified.
+    Private Friend Protected Field3 As Integer
+            ~~~~~~
+BC30176: Only one of 'Public', 'Private', 'Protected', 'Friend', or 'Protected Friend' can be specified.
+    Friend Protected Private Field4 As Integer
+                     ~~~~~~~
+</errors>)
```

Comment:
Test `Private Protected Public` so it's clear we're not just testing existing `Private` + `Protected Friend` behavior.